optional/compatible_abc.py

```diff
@@ -0,0 +1,4 @@
+from abc import ABCMeta
+
+
+CompatibleABC = ABCMeta('ABC', (object,), {'__slots__': ()})
```

Comment:
I think we need an inline comment here `# for python 2 and 3 compatibility`

Reply:
I thought moving it out into its own file did enough. =\

Reply:
I added a package docstring.